crates/wasi/src/preview1.rs

```diff
@@ -2321,6 +2321,29 @@ impl wasi_snapshot_preview1::WasiSnapshotPreview1 for WasiP1Ctx {
             // Indefinite sleeping is not supported in preview1.
             return Err(types::Errno::Inval.into());
         }
+
+        if nsubscriptions == 1 {
+            let sub = subs.read()?;
+            if let types::SubscriptionU::Clock(clocksub) = sub.u {
+                if !clocksub
+                    .flags
+                    .contains(types::Subclockflags::SUBSCRIPTION_CLOCK_ABSTIME)
+                {
+                    std::thread::sleep(std::time::Duration::from_nanos(clocksub.timeout));
```

Comment:
Can this use `self.ctx().allow_blocking_current_thread` to dispatch on whether to call `std::thread::sleep` or do an async tokio-based sleep?

Reply:
Hi @alexcrichton ,  
Yes, we can use it. Take a look at the updated change. If `self.ctx().allow_blocking_current_thread` is false, we just skip calling `std::thread::sleep`, and then do the rest of `poll_oneoff` to do an async tokio-based sleep.